src/targets/__tests__/symbolCollector.test.ts

```diff
@@ -0,0 +1,138 @@
+import { withTempDir } from '../../utils/files';
+import { NoneArtifactProvider } from '../../artifact_providers/none';
+import { checkExecutableIsPresent, spawnProcess } from '../../utils/system';
+import { SymbolCollector, SYM_COLLECTOR_BIN_NAME } from '../symbolCollector';
+
+jest.mock('../../utils/files');
+jest.mock('../../utils/system');
+jest.mock('fs', () => {
+  const original = jest.requireActual('fs');
+  return {
+    ...original,
+    promises: {
+      mkdir: jest.fn(() => {
+        /** do nothing */
+      }),
+    },
+  };
+});
+
+const customConfig = {
+  batchType: 'batchType',
+  bundleIdPrefix: 'bundleIdPrefix-',
+};
+
+function getSymbolCollectorInstance(
+  config: Record<string, unknown> = { testKey: 'testVal' }
+): SymbolCollector {
+  return new SymbolCollector(
+    {
+      name: 'symbol-collector',
+      ...config,
+    },
+    new NoneArtifactProvider()
+  );
+}
+
+describe('target config', () => {
+  test('symbol collector not present in path', () => {
+    (checkExecutableIsPresent as jest.MockedFunction<
+      typeof checkExecutableIsPresent
+    >).mockImplementationOnce(() => {
+      throw new Error('Checked for executable');
+    });
+
+    expect(getSymbolCollectorInstance).toThrowErrorMatchingInlineSnapshot(
+      `"Checked for executable"`
+    );
+    expect(checkExecutableIsPresent).toHaveBeenCalledTimes(1);
+    expect(checkExecutableIsPresent).toHaveBeenCalledWith(
+      SYM_COLLECTOR_BIN_NAME
+    );
+  });
+
+  test('config missing', () => {
+    (checkExecutableIsPresent as jest.MockedFunction<
+      typeof checkExecutableIsPresent
+    >) = jest.fn();
+
+    expect(getSymbolCollectorInstance).toThrowErrorMatchingInlineSnapshot(
+      '"The required `batchType` parameter is missing in the configuration file. ' +
+        'See the documentation for more details."'
+    );
+    expect(checkExecutableIsPresent).toHaveBeenCalledTimes(1);
+    expect(checkExecutableIsPresent).toHaveBeenCalledWith(
+      SYM_COLLECTOR_BIN_NAME
+    );
```

Comment:
I'd lift these checks into their own case as this doesn't seem relevant to the case we are testing.